File: leetcode/2586.统计范围内的元音字符串数.go
func vowelStrings(words []string, left int, right int) int {
    ans := 0
    yuan := map[byte]bool{'a':true, 'e':true, 'i':true, 'o':true, 'u':true}
    for _, word := range(words[left: right + 1]) {
        if yuan[word[0]] && yuan[word[len(word) - 1]] {
            ans += 1
        }
    }
    return ans
}

File: leetcode/2586.统计范围内的元音字符串数.py
class Solution:
    def vowelStrings(self, words: List[str], left: int, right: int) -> int:
        ans = 0
        y = ['a','e','i','o','u']
        for i, word in enumerate(words):
            if word[0] in y and word[-1] in y and i<=right and i>=left:
                ans += 1
        return ans